src/DDTrace/Integrations/Yii/V2/YiiIntegrationLoader.php

```diff
@@ -37,6 +37,22 @@ function (SpanData $span) use ($service) {
             }
         );
 
+        // We assume the first controller is the one to assign to app.endpoint
+        $first_controller = null;
+        \dd_trace_method(
+            'yii\base\Module',
+            'createController',
+            function (SpanData $span, $args, $retval, $ex) use (&$first_controller) {
+                if (!$ex && isset($args[0], $retval) && \is_array($retval) && !empty($retval)) {
+                    $first_controller = $retval[0];
+                    if ($this->requestedRoute === $args[0]) {
+                        $first_controller = $retval[0];
+                    }
```

Comment:
I think these lines have no effect.

Reply:
The `$first_controller` is captured by reference, and is then used in `yii\base\Controller::runAction`.

Reply:
What I mean is that line 49 has no effects as line 47 (identical) has already been executed

Reply:
Ah, I see. It's line 47 that should be removed. Oops!

Reply:
👍 now it makes sense 😄